filesystem/fat32/fat32.go

```diff
@@ -810,21 +810,21 @@ func (fs *FileSystem) allocateSpace(size uint64, previous uint32) ([]uint32, err
 		}
 	}
 
-	// did we allocate them all?
 	if len(allocated) < extraClusterCount {
 		return nil, errors.New("No space left on device")
 	}
-	// mark last allocated one as EOC
-	lastAlloc := len(allocated) - 1
 
-	// extend the chain and fill them in
-	if previous > 0 {
-		allClusters[previous] = allocated[0]
+	for i, next := range allocated {
+		if i == 0 {
+			continue
+		}
+		cur := allocated[i-1]
+		fs.table.clusters[cur] = next
 	}
-	for i := 0; i < lastAlloc; i++ {
-		allClusters[allocated[i]] = allocated[i+1]
+	fs.table.clusters[allocated[len(allocated)-1]] = fs.table.eocMarker
+	if len(clusters) > 0 {
+		fs.table.clusters[clusters[len(clusters)-1]] = allocated[0]
```

Comment:
It looks like you are rejiggering the tables here? Why not just append it to the end?